test/cypress/README.md

```diff
@@ -0,0 +1,66 @@
+# Functional Testing with Cypress
+
+In order to ensure that as we upgrade our wagtail/django dependencies our backend python code is still functional from a frontend perspective we are integrating a new browser testing tool [Cypress](https://www.cypress.io).
+
+## Installing Cypress
+
+We have included Cypress as a dependency of this project, and so the only steps that need to be taken are doing a fresh `yarn` if you haven't already.
+
+## Running Cypress
+
+We support both a headless docker container to execute our cypress tests as well as the already installed desktop application that comes packaged with cypress. 
+
+* To run the docker container execute `docker-compose -f docker-compose.e2e.yml run e2e`
+
+* To run the desktop cypress aplication execute `yarn run cypress open` and select your test file to execute.
+
+## Writing Cypress Tests
+
+When developing new tests for Cypress it is important to consider what the test is trying to accomplish. We want to ensure that we are not polluting our cypress tests with things that can be tested at another level, we are seeking to only test the integration aspect of our UI hosted via Wagtail/Django and custom python code within our backend.
+
+When adding a test it is often helpful to separate the arrange/act code from the actual assertions in order to improve the readability of our testing code. To do this we have adopted the page model of testing, where we define a page within the application and the methods of interacting with the page separate from the test file itself where we define the assertions. 
+
+For example consider the ConsumerTools Page:
+
+```javascript
+export default class ConsumerTools {
+    
+    constructor() {}
+    
+    open() {
+        cy.visit('/consumer-tools/');
+    }
+
+    signUp(email) {
+        cy.get('.o-form__email-signup').within(() => {
+            cy.get('input:first').type(email);
+            cy.get('button:first').click();
+        });
+    }
+    successNotification() {
+        return cy.get('.m-notification_message');
+    }
+}
+```
+
+Notice how this class defines functions to retrieve and modify elements on the page but in a more human readable manner. This allows our test file for consumer tools to look like:
+
+```javascript
+import ConsumerTools from '../pages/consumer_tools';
+
+let page = new ConsumerTools();
+
+describe('Consumer Tools', () => {
+    it('Should have an email sign up', () => {
+        // Arrange
+        page.open();
+        //Act
+        page.signUp('testing@cfpb.gov');
+        // Assert
+        page.successNotification().should('exist');
+        page.successNotification().contains('Your submission was successfully received.')
+    });
+});
+```
+
+Overall it lets our tests show what is intended to be happening on a page without showing the more technical side of how we reference and interact with elements.
```

Comment:
Can you include a brief mention of handling network requests somewhere in this doc? Just a sentence that links people to [this page](https://docs.cypress.io/guides/guides/network-requests.html).

Reply:
Given the current focus on functional testing, I'm not sure we want to encourage the use of manipulating network requests and responses. In this PR I'm doing it mainly because we are lacking some configuration but down the line the current goal is to actual communicate with our backend python code for all the tests. IF we end up preferring our cypress testing to the existing browser based tests we might need to re-evaluate our stance on this.